internal/services/network/subnet_resource.go

```diff
@@ -214,10 +249,77 @@ func resourceSubnetCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 
 	// To enable private endpoints you must disable the network policies for the subnet because
 	// Network policies like network security groups are not supported by private endpoints.
-	privateEndpointNetworkPolicies := d.Get("enforce_private_link_endpoint_network_policies").(bool)
-	privateLinkServiceNetworkPolicies := d.Get("enforce_private_link_service_network_policies").(bool)
-	properties.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(privateEndpointNetworkPolicies))
-	properties.PrivateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(privateLinkServiceNetworkPolicies))
+	var privateEndpointNetworkPolicies network.VirtualNetworkPrivateEndpointNetworkPolicies
+	var privateLinkServiceNetworkPolicies network.VirtualNetworkPrivateLinkServiceNetworkPolicies
+
+	if features.FourPointOhBeta() {
+		privateEndpointNetworkPoliciesRaw := d.Get("private_endpoint_network_policies_enabled").(bool)
+		privateLinkServiceNetworkPoliciesRaw := d.Get("private_link_service_network_policies_enabled").(bool)
+
+		privateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetNetworkPolicy(privateEndpointNetworkPoliciesRaw))
+		privateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetNetworkPolicy(privateLinkServiceNetworkPoliciesRaw))
+	} else {
+		var enforceOk bool
+		var enforceServiceOk bool
+		var enableOk bool
+		var enableServiceOk bool
+		var enforcePrivateEndpointNetworkPoliciesRaw bool
+		var enforcePrivateLinkServiceNetworkPoliciesRaw bool
+		var privateEndpointNetworkPoliciesRaw bool
+		var privateLinkServiceNetworkPoliciesRaw bool
+
+		// Set the legacy default value since they are now computed optional
+		privateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPoliciesEnabled
+		privateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPoliciesEnabled
+
+		// This is the only way I was able to figure out if the fields are actually in the config or not,
+		// which is needed here because these are all now optional computed fields...
+		enforcePrivateEndpointNetworkPoliciesCty := d.GetRawConfig().AsValueMap()["enforce_private_link_endpoint_network_policies"]
+		enforcePrivateLinkServiceNetworkPoliciesCty := d.GetRawConfig().AsValueMap()["enforce_private_link_service_network_policies"]
+		privateEndpointNetworkPoliciesCty := d.GetRawConfig().AsValueMap()["private_endpoint_network_policies_enabled"]
+		privateLinkServiceNetworkPoliciesCty := d.GetRawConfig().AsValueMap()["private_link_service_network_policies_enabled"]
+
+		if !enforcePrivateEndpointNetworkPoliciesCty.IsNull() {
```

Comment:
whilst it's fine to use this method here, can we wrap this in a helper method, since that'd make this easier when we refactor this to use Framework later on, something along the lines of:

```
if pluginsdk.IsExplicitlyNullInConfig(d, "private_link_service_network_policies_enabled") {
  //
}
```

Reply:
Fixed